metric_learn/lsml.py

```diff
@@ -31,18 +33,19 @@ def __init__(self, tol=1e-3, max_iter=1000, prior=None, verbose=False):
         guess at a metric [default: inv(covariance(X))]
     verbose : bool, optional
         if True, prints information while learning
+    preprocessor : array-like, shape=(n_samples, n_features) or callable
+        The preprocessor to call to get tuples from indices. If array-like,
+        tuples will be formed like this: X[indices].
     """
     self.prior = prior
     self.tol = tol
     self.max_iter = max_iter
     self.verbose = verbose
+    super(_BaseLSML, self).__init__(preprocessor)
 
   def _prepare_quadruplets(self, quadruplets, weights):
-    # for now we check_array and check_tuples but we should only
-    # check_tuples in the future (with enhanced check_tuples)
-    quadruplets = check_array(quadruplets, accept_sparse=False,
-                              ensure_2d=False, allow_nd=True)
-    quadruplets = check_tuples(quadruplets)
+    quadruplets = self._prepare_inputs(quadruplets,
+                                       type_of_inputs='tuples')
```

Comment:
Maybe move possible values of `type_of_inputs` to a constant class / enum.  Otherwise it's easy to make mistakes?

Reply:
I didn't think of it,  I agree it would be cleaner. But is there a way to do this without the need to import the class/enum every time we want to fill the argument ? Like if we create TYPE_OF_INPUTS = Enum('TYPE_OF_INPUTS', 'classic tuples') in _util, anyone who would like to fill the argument "type_of_inputs=" of `check_input` would have to import `TYPE_OF_INPUTS` from `metric_learn._util` right ? This could add a bit of complexity for users used to put strings in arguments like I think is the case in many scikit-learn estimators (like method='svd' etc)? And if as it is in the docstring it will help them to put the right argument. But in this case maybe we should raise an error if the given string is not in {'tuples', 'classic'}, reminding the allowed strings ?

Reply:
It seems that `_prepare_inputs` isn't user-facing. So long as users won't need to (commonly) pass an enum as an argument in the public API, I'm +1 for using an enum.

In any case, we should definitely raise an error if an unexpected value is passed.